### PR TITLE
feat: add events and wait option

### PR DESCRIPTION
**Pull request by scottd018**

### CODE REVIEW

File: pkg/status/events.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+			"%s child resource '%s' managed by parent resource '%s'",
+			event.String(),
+			getMessageString(child),
+			getMessageString(parent),
+		),
+	)
+}
+
+// getMessageString gets the message string for an object.  The message string is the message that is
+// displayed when a resource is actioned upon.
```
Comment (lander2k2) [inline pkg/status/events.go]:
```suggestion
// displayed when a resource is acted upon.
```

File: pkg/controller/phases/option.go [hunk 2]
```diff
@@ -7,10 +7,46 @@ import ctrl "sigs.k8s.io/controller-runtime"
...
+
 // WithCustomRequeueResult allows you to define a custom result for a phase when it is requeued,
 // this allows for custom requeue time backoffs.
 func WithCustomRequeueResult(requeueResult ctrl.Result) PhaseOption {
 	return func(p *Phase) {
 		p.requeueResult = &requeueResult
 	}
 }
+
+// WithWait adds the ResourceOptionWithWait resource option to the phase.
```
Comment (lander2k2) [inline pkg/controller/phases/option.go]: This function's doc comment doesn't match the func

**Review by lander2k2 — CHANGES_REQUESTED** [review]

File: pkg/controller/phases/option.go (hunk 2 above)
Comment (scottd018) [reply pkg/controller/phases/option.go]: Resolved in latest commit.

File: pkg/resources/admission_webhooks.go [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+const (
+	MutatingWebhookConfigurationKind    = "MutatingWebhookConfiguration"
+	MutatingWebhookConfigurationVersion = "admissionregistration.k8s.io/v1"
+
+	ValidatingWebhookConfigurationKind    = "ValidatingWebhookConfiguration"
+	ValidatingWebhookConfigurationVersion = "admissionregistration.k8s.io/v1"
+)
+
+type MutatingWebhookConfigurationResource struct {
```
Comment (lander2k2) [inline pkg/resources/admission_webhooks.go:23]: Can we add a doc comment here?

File: pkg/resources/admission_webhooks.go [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+	Object     admissionv1.MutatingWebhookConfiguration
+	Reconciler workload.Reconciler
+	Request    *workload.Request
+}
+
+type ValidatingWebhookConfigurationResource struct {
```
Comment (lander2k2) [inline pkg/resources/admission_webhooks.go:31]: Can we add a doc comment here?

File: pkg/resources/endpoints.go [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	EndpointsKind    = "Endpoints"
+	EndpointsVersion = "v1"
+)
+
+type EndpointsResource struct {
```
Comment (lander2k2) [inline pkg/resources/endpoints.go:17]: Can we add a doc comment here?

**Review by lander2k2 — CHANGES_REQUESTED** [review]

File: pkg/resources/admission_webhooks.go (hunk 3 above)
Comment (scottd018) [reply pkg/resources/admission_webhooks.go:23]: Fixed in [46cbf84](https://github.com/nukleros/operator-builder-tools/pull/5/commits/46cbf8426e1841f3bd28610071247ea56c6ab8c8)

File: pkg/resources/admission_webhooks.go (hunk 4 above)
Comment (scottd018) [reply pkg/resources/admission_webhooks.go:31]: Fixed in [46cbf84](https://github.com/nukleros/operator-builder-tools/pull/5/commits/46cbf8426e1841f3bd28610071247ea56c6ab8c8)

**Review by lander2k2 — APPROVED** [review]

**Pull request merged**